diff --git a/python/sglang/srt/layers/layernorm.py b/python/sglang/srt/layers/layernorm.py
index c2a0ba5..c5b92ff 100644
--- a/python/sglang/srt/layers/layernorm.py
+++ b/python/sglang/srt/layers/layernorm.py
@@ -107,6 +107,7 @@ if _is_cuda or _is_xpu or _is_musa:
 _has_aiter_layer_norm = False
 _has_vllm_rms_norm = False
 _has_rocm_triton_gemma_rms_norm = False
+_disable_vllm_rmsnorm = get_bool_env_var("SGLANG_DISABLE_VLLM_RMSNORM")
 if _use_aiter:
     import aiter as _aiter
     from aiter import layernorm2d_fwd as layer_norm
@@ -115,7 +116,7 @@ if _use_aiter:
 
     _has_aiter_layer_norm = True  # aiter provides the layer_norm functions
     _has_vllm_rms_norm = True  # aiter provides the rms_norm functions
-elif _is_hip:
+elif _is_hip and not _disable_vllm_rmsnorm:
     try:
         from vllm._custom_ops import fused_add_rms_norm, rms_norm
 
